projects/observability/src/shared/dashboard/widgets/topology/node/box/entity-node-box-renderer.service.ts

```diff
@@ -101,13 +107,21 @@ export abstract class EntityNodeBoxRendererService implements TopologyNodeRender
     domElementRenderer: Renderer2
   ): void {
     const elementSelection = this.d3Utils.select(element, domElementRenderer);
-    const metricSpecifications = state.dataSpecifiers?.map(specifier => specifier.value);
 
-    this.updateNodeMetric(
-      elementSelection,
-      getLatencyMetric(node.data, metricSpecifications),
-      getErrorPercentageMetric(node.data, metricSpecifications)
+    const primaryMetric = this.topologyDataSourceModelPropertiesService.getPrimaryNodeMetric();
```

Comment:
Can probably assign this in the constructor and assign it to corresponding readonly properties

Reply:
I tried that, But it did not work, since the the data is being fetched asynchronous from API and then fed to the service, so assigning into constructor would not work.

Reply:
hmm.. Okay i can pull it up and check